### PR TITLE
API-32268: Benefits Intake Monthly Report Bug Fixes and Spec Additions

### PR DESCRIPTION
**Pull request by kristen-brown**
## Summary
This PR fixes a couple of bugs in the `VBADocuments` (Benefits Intake) monthly report, adds some specs for the reporting feature, and refactors a few areas of the `VBADocuments` specs and factories. The majority of the changes are in the `modules/vba_documents/spec` directory.

Bugs fixed:
1. Monthly report now returns the total submission count rather than page count total in the "Prior Twelve Months - Page Counts and Upload Sizes in MB" section of the report. It previously returned the total submission count, so this change just puts it back to how it was before the report refactoring took place.
2. If there are `VBADocuments::UploadSubmission` records with a `consumer_name` of `nil` (e.g. forensics records), these are excluded from the report and the report generation no longer errors when trying to sort the `consumer_names`.

Misc. improvements:
1. Various spec files were added, which cover the reporting feature. 
2. A `VBADocuments::MonthlyStat` factory was added. 
3. The `VBADocuments::UploadSubmission` factory was updated to use `SecureRandom.uuid` to generate the `guid`. Prior to this change, if a developer wanted to create multiple `VBADocuments::UploadSubmission` records from the factory, they would error with "record not unique." This change also allowed a few spec files to be updated to remove `guid: SecureRandom.uuid`, which was used to get around the prior error.
4. The specs for the `get_recipients` mailer method were updated to use the `with_settings` helper and edited for clarity.

## Related issue(s)
[API-32268](https://jira.devops.va.gov/browse/API-32268)

## Testing done
I generated the monthly report locally to ensure that the output is expected after the bug fixes were put in place. This PR also adds more specs for the monthly reporting feature which cover the prior bug scenarios.

## Screenshots
None

## What areas of the site does it impact?
This PR impacts the `vba_documents` module only, and mostly the spec files, as well as the monthly reporting feature.

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation) – N/A
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable) – N/A
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature – N/A

## Requested Feedback
No specific feedback requested.

### CODE REVIEW

File: modules/vba_documents/app/sidekiq/vba_documents/report_monthly_submissions.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require 'sidekiq'
+require 'sidekiq/monitored_worker'
```
Comment (kristen-brown) [inline modules/vba_documents/app/sidekiq/vba_documents/report_monthly_submissions.rb:4]: Without this line, the specs wouldn't run successfully. This was a prior oversight.

File: modules/vba_documents/app/views/vba_documents/monthly_report_mailer/monthly_report.html.erb [hunk 2]
```diff
@@ -75,7 +75,7 @@
       <tr>
         <td><%= month_stats.year %></td>
         <td><%= Date::MONTHNAMES[month_stats.month] %></td>
-        <td style="text-align: right"><%= month_stats.stats['page_count_stats']['total'] %></td>
+        <td style="text-align: right"><%= month_stats.stats['summary_stats']['total'] %></td>
```
Comment (kristen-brown) [inline modules/vba_documents/app/views/vba_documents/monthly_report_mailer/monthly_report.html.erb:78]: This fixes Bug #1 explained in the PR description.

File: modules/vba_documents/lib/vba_documents/monthly_stats_generator.rb [hunk 3]
```diff
@@ -141,6 +141,7 @@ def end_date
     def records_in_date_range
       @records_in_date_range ||= VBADocuments::UploadSubmission
                                  .where('created_at >= ? and created_at <= ?', start_date, end_date)
+                                 .where.not(consumer_name: nil)
```
Comment (kristen-brown) [inline modules/vba_documents/lib/vba_documents/monthly_stats_generator.rb:144]: These changes fixe Bug #2 explained in the PR description.

File: modules/vba_documents/spec/factories/upload_submissions.rb [hunk 4]
```diff
@@ -20,7 +20,7 @@
   }.freeze
 
   factory :upload_submission, class: 'VBADocuments::UploadSubmission' do
-    guid { 'f7027a14-6abd-4087-b397-3d84d445f4c3' }
+    guid { SecureRandom.uuid }
```
Comment (kristen-brown) [inline modules/vba_documents/spec/factories/upload_submissions.rb:23]: This allows a developer to create multiple `UploadSubmission` records from the factory in the same spec without generating a "guid not unique" error.

File: modules/vba_documents/spec/factories/upload_submissions.rb [hunk 5]
```diff
@@ -58,15 +55,24 @@
...
+    trait :skip_record_status_change_callback do
+      after(:build) do |foo|
+        class << foo
+          def record_status_change
+            true
+          end
+        end
+      end
+    end
   end
```
Comment (kristen-brown) [inline modules/vba_documents/spec/factories/upload_submissions.rb:68]: This trait was added to skip some unhelpful automatic behavior in the `monthly_stats_generator_spec` which overrode the status transition times that were already set in the specs, causing the transition times to change continuously and the specs to fail. It only changes the behavior if the developer chooses to intentionally use this trait when creating records in their spec.

**Comment by kristen-brown** [comment]
This PR is above the 500 line limit, but most of the lines changed are data config for specs. Once this is approved, I'm going to request an exception for the Danger error.

**Review by mathisto — APPROVED** [review]

**Review by michelpmcdonald — APPROVED** [review]

**Pull request merged**